lib/msf/core/exploit/remote/kerberos/client/as_response.rb

```diff
@@ -40,6 +41,12 @@ def extract_logon_time(res, key)
 
               auth_time.to_i
             end
+
+            # @param [Rex::Proto::Kerberos::Model::KdcResponse] asrep The krb5 asrep response
+            # @return [String] A valid string format which can be cracked offline
+            def format_as_rep_to_john_hash(asrep)
+              "$krb5asrep$#{asrep.enc_part.etype}$#{asrep.cname.name_string.join('/')}@#{asrep.ticket.realm}:#{asrep.enc_part.cipher[0...16].unpack1('H*')}$#{asrep.enc_part.cipher[16..].unpack1('H*')}"
```

Comment:
doc link for this might be useful for future travellers so they don't have a panic attack :upside_down_face:

Reply:
I don't believe there's an rfc doc or anything similar to that, I believe it's just an arbitrary format that john/hashcat supports